src/domain/usecases/SendUserInfoNotificationsUseCase.ts

```diff
@@ -0,0 +1,61 @@
+import { RecipientRepository } from "domain/repositories/RecipientRepository";
+import { NotificationsRepository } from "domain/repositories/NotificationsRepository";
+import { UserInfoNotificationsRepository } from "domain/repositories/UserInfoNotificationsRepository";
+import { Path } from "domain/entities/Base";
+import log from "utils/log";
+import _ from "lodash";
+import { UsernameEmail } from "domain/entities/Notification";
+import { promiseMap } from "data/dhis2-utils";
+
+interface Options {
+    url: string;
+    userDetailsFile: Path;
+    emailContentFile: Path;
+}
+
+interface UserDetails extends UsernameEmail {
+    password: string;
+}
+
+export class SendUserInfoNotificationsUseCase {
+    constructor(
+        private userInfoNotificationRepository: UserInfoNotificationsRepository,
+        private recipientRepository: RecipientRepository,
+        private notificationsRepository: NotificationsRepository
+    ) {}
+
+    async execute(options: Options) {
+        log.debug(`Get User Info and Email Content`);
+        const userInfoNotification = await this.userInfoNotificationRepository.get(
+            options.userDetailsFile,
+            options.emailContentFile
+        );
+
+        log.debug(`Get emails for each user in CSV file from DHIS2`);
+        const usernames = _(userInfoNotification.userInfos)
+            .map(({ username }) => username)
+            .compact()
+            .value();
+        const recipients = await this.recipientRepository.getByUsernames(usernames);
+
+        log.debug(`Send user info email to: ${recipients.join(", ")}`);
+        const recipientsMap = recipients.reduce<{ [index: string]: UsernameEmail }>((acc, curr) => {
+            acc[curr.username] = curr;
+            return acc;
+        }, {});
```

Comment:
A reduce can typically used with other lodash abstractions (groupBy, fromPairs, and so on). In this case, you are doing an indexing/keying, there are an abstraction for that -> `const recipientsMap = _.keyBy(recipients, recipient => recipient.username)`